flang/lib/Lower/Allocatable.cpp

```diff
@@ -799,17 +799,40 @@ static void postDeallocationAction(Fortran::lower::AbstractConverter &converter,
     Fortran::lower::attachDeclarePostDeallocAction(converter, builder, sym);
 }
 
+static mlir::Value genCudaDeallocate(fir::FirOpBuilder &builder,
+                                     mlir::Location loc,
+                                     const fir::MutableBoxValue &box,
+                                     ErrorManager &errorManager,
+                                     const Fortran::semantics::Symbol &sym) {
+  fir::CUDADataAttributeAttr cudaAttr =
+      Fortran::lower::translateSymbolCUDADataAttribute(builder.getContext(),
+                                                       sym);
+  mlir::Value errmsg =
+      mlir::isa<fir::AbsentOp>(errorManager.errMsgAddr.getDefiningOp())
```

Comment:
This check is different from https://github.com/llvm/llvm-project/pull/88980/files#diff-96be36dcf5ea9de4eb09e8224e069245ad5a3bd4cc8cae81d429f7e6cdaa68ddR735 - should they be consistent?

Reply:
The reason it is different is that this function is not a member function from the `AllocateStmtHelper` so there is no access to `errMsgExpr`. I can update the check in the genCudaAllocate function to match this one.

Reply:
Oh, I see. It is good as-is.